### PR TITLE
Added electronic journal to the transaction search

### PR DESCRIPTION
**Pull request by blubaugha**
### Issues Fixed
[#2243](https://github.com/JumpMind/commerce/issues/2243)

### Summary
This adds an **Electronic Journal**, to the transaction search screen, that allows users to search all transaction types. Also, the **View Details** button on `POST_VOID` transaction types will navigate to the original voided transaction.

### Parameter Changes
There is a configuration option to enable/disable this feature. It can be used to enable this for the BO and disable on the register.
```
openpos:
  transactionSearch:
    DOC__electronicJournal: "Enables the electronic journal."
    enableElectronicJournal: true
```

### Screenshots
**Post Void -> Navigates to original voided transaction**

trans-search--desktop-post-void

**Disable Electronic Journal**

trans-search--disable-electronic-journal

**Desktop**

trans-search--desktop

**ELO POS - Landscape**

trans-search--elopos-landscape

**ELO POS - Portrait**

trans-search--elopos-portrait

**iPad Pro - Landscape**

trans-search--ipadpro-landscape

**iPad Pro - Portrait**

trans-search--ipadpro-portrait

**iPad - Landscape**

trans-search--ipad-landscape

**iPad - Portrait**

trans-search--ipad-portrait

### CODE REVIEW

File: openpos-flow/src/main/java/org/jumpmind/pos/core/ui/IconType.java [hunk 1]
```diff
@@ -60,13 +60,15 @@
     public static final String Journal = "Journal"; //book
     public static final String KebabMenu = "KebabMenu"; //more_vert
     public static final String Lease = "Lease"; // account_balance
+    public static final String LightSolid = "LightSolid"; //ws_incandescent_up
```
Comment (dankaste) [inline openpos-flow/src/main/java/org/jumpmind/pos/core/ui/IconType.java:63]: We don't need this either. As mentioned above sending over the svg file name to the client will build the correct content url. If we want to add a layer of abstraction there is an IconConfig object that can be read in through configuration.

**Review by dankaste — APPROVED** [review]

File: openpos-service/src/main/java/org/jumpmind/pos/service/util/DateUtils.java [hunk 2]
```diff
@@ -68,5 +68,17 @@ public static long daysBetween(Date date1, Date date2) {
...
+    public static String changeFormat(String value, String existingFormat, String newFormat) {
+        if(StringUtils.isBlank(value)) {
+            return value;
+        }
+
+        try {
+            Date date = new SimpleDateFormat(existingFormat).parse(value);
+            return new SimpleDateFormat(newFormat).format(date);
+        } catch (ParseException e) {
+            return value;
```
Comment (jrmihalick) [inline openpos-service/src/main/java/org/jumpmind/pos/service/util/DateUtils.java:81]: See comment on #945

**Review by jrmihalick — CHANGES_REQUESTED** [review]

File: openpos-service/src/main/java/org/jumpmind/pos/service/util/DateUtils.java (hunk 2 above)
Comment (blubaugha) [reply openpos-service/src/main/java/org/jumpmind/pos/service/util/DateUtils.java:85]: Fixed

Comment (jrmihalick) [reply openpos-service/src/main/java/org/jumpmind/pos/service/util/DateUtils.java:85]: See my follow up comment on #945

**Review by jrmihalick — CHANGES_REQUESTED** [review]

File: openpos-service/src/main/java/org/jumpmind/pos/service/util/DateUtils.java (hunk 2 above)
Comment (blubaugha) [reply openpos-service/src/main/java/org/jumpmind/pos/service/util/DateUtils.java:85]: Fixed for real this time 😄

**Review by jrmihalick — APPROVED** [review]

**Pull request merged**